common/djangoapps/util/milestones_helpers.py

```diff
@@ -86,33 +87,46 @@ def set_prerequisite_courses(course_key, prerequisite_course_keys):
+def get_pre_requisite_courses_not_completed(user, enrolled_courses):  # pylint: disable=invalid-name
     """
-    It would make dict of prerequisite courses not completed by user among courses
-    user has enrolled in. It calls the fulfilment api of milestones app and
-    iterates over all fulfilment milestones not achieved to make dict of
-    prerequisite courses yet to be completed.
+    Makes a dict mapping courses to their incomplete requirements using the
+    fulfillment API of the milestones app.
+
+    Arguments:
+        user (User): the user for whom we are checking prerequisites.
+        enrolled_courses (CourseKey): a list of keys for the courses to be
+            checked. The given user must be enrolled in all of these courses.
+
+    Returns:
+        dict[CourseKey: dict[
+            'courses': list[dict['key': CourseKey, 'display': str]]
+        ]]
+        If a course has no incomplete prerequisites, it will be excluded from the
+        dictionary.
     """
+    if not settings.FEATURES.get('ENABLE_PREREQUISITE_COURSES', False):
+        return {}
+
+    from milestones import api as milestones_api
     pre_requisite_courses = {}
-    if settings.FEATURES.get('ENABLE_PREREQUISITE_COURSES', False):
-        from milestones import api as milestones_api
-        for course_key in enrolled_courses:
-            required_courses = []
-            fulfilment_paths = milestones_api.get_course_milestones_fulfillment_paths(course_key, {'id': user.id})
-            for milestone_key, milestone_value in fulfilment_paths.items():  # pylint: disable=unused-variable
-                for key, value in milestone_value.items():
-                    if key == 'courses' and value:
-                        for required_course in value:
-                            required_course_key = CourseKey.from_string(required_course)
-                            required_course_descriptor = modulestore().get_course(required_course_key)
-                            required_courses.append({
-                                'key': required_course_key,
-                                'display': get_course_display_name(required_course_descriptor)
-                            })
-
-            # if there are required courses add to dict
-            if required_courses:
-                pre_requisite_courses[course_key] = {'courses': required_courses}
+
+    for course_key in enrolled_courses:
+        required_courses = []
+        fulfillment_paths = milestones_api.get_course_milestones_fulfillment_paths(course_key, {'id': user.id})
+        for __, milestone_value in fulfillment_paths.items():
+            for key, value in milestone_value.items():
+                if key == 'courses' and value:
+                    for required_course in value:
+                        required_course_key = CourseKey.from_string(required_course)
+                        required_course_overview = CourseOverview.get_from_id(required_course_key)
```

Comment:
Note to reviewers: `modulestore().get_course` was changed to `CourseOverview.get_from_id`.